Tests/UIKitTests/UIButtonExtensionsTests.swift

```diff
@@ -144,35 +144,93 @@ final class UIButtonExtensionsTests: XCTestCase {
         XCTAssertEqual(button.titleForSelected, title)
     }
 
-    func testCenterTextAndImage() {
+    func testSetImageLeftText() {
         let button = UIButton(frame: CGRect(x: 10, y: 10, width: 100, height: 100))
         let image = UIImage(color: .green, size: CGSize(width: 10, height: 10))
         button.setTitleForAllStates("Title")
         button.setImageForAllStates(image)
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 10)
 
         XCTAssertNotNil(button.imageView)
         XCTAssertNotNil(button.titleLabel)
 
         let spacing: CGFloat = 20
 
         // Image on the left of text
-        button.centerTextAndImage(spacing: spacing)
-        var imageFrame = button.imageView!.frame
-        var titleFrame = button.titleLabel!.frame
+        button.setImageLeftOfText(spacing: spacing)
+        let imageFrame = button.imageView!.frame
+        let titleFrame = button.titleLabel!.frame
 
-        XCTAssert(titleFrame.midX > imageFrame.midX)
+        XCTAssert(titleFrame.minX > imageFrame.maxX)
         XCTAssertEqual(titleFrame.midY, imageFrame.midY, accuracy: 1.0)
         XCTAssertEqual(titleFrame.minX - spacing, imageFrame.maxX, accuracy: 1.0)
+    }
+
+    func testSetImageRightText() {
+        let button = UIButton(frame: CGRect(x: 10, y: 10, width: 100, height: 100))
+        let image = UIImage(color: .green, size: CGSize(width: 10, height: 10))
+        button.setTitleForAllStates("Title")
+        button.setImageForAllStates(image)
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 10)
+
+        XCTAssertNotNil(button.imageView)
+        XCTAssertNotNil(button.titleLabel)
+
+        let spacing: CGFloat = 20
+
+        // Image on the right of text
+        button.setImageRightOfText(spacing: spacing)
+        let imageFrame = button.imageView!.frame
+        let titleFrame = button.titleLabel!.frame
+
+        XCTAssert(titleFrame.minX < imageFrame.maxX)
+        XCTAssertEqual(titleFrame.midY, imageFrame.midY, accuracy: 1.0)
+        XCTAssertEqual(imageFrame.minX - spacing, titleFrame.maxX, accuracy: 1.0)
+    }
+
+    func testSetImageAboveText() {
```

Comment:
```Swift
    func testSetImageAboveText() {
        let button = UIButton()
        let image = UIImage(color: .green, size: CGSize(width: 10, height: 10))
        button.setTitleForAllStates("Title")
        button.setImageForAllStates(image)
        button.titleLabel?.font = UIFont.systemFont(ofSize: 10)

        XCTAssertNotNil(button.imageView)
        XCTAssertNotNil(button.titleLabel)

        let spacing: CGFloat = 20
        button.setImageAboveText(spacing: spacing)
        let imageFrame = button.imageView!.frame
        let titleFrame = button.titleLabel!.frame

        XCTAssertEqual(titleFrame.midX, imageFrame.midX, accuracy: 1.0)
        XCTAssertEqual(min(titleFrame.minX, imageFrame.minX), button.bounds.minX, accuracy: 1.0)
        XCTAssertEqual(max(titleFrame.maxX, imageFrame.maxX), button.bounds.maxX, accuracy: 1.0)
        XCTAssertEqual(titleFrame.minY - spacing, imageFrame.maxY, accuracy: 1.0)
        XCTAssertEqual(imageFrame.minY, button.bounds.minY, accuracy: 1.0)
        XCTAssertEqual(titleFrame.maxY, button.bounds.maxY, accuracy: 1.0)
    }
```

Reply:
According to the above description, I don’t think it needs to be changed.

Reply:
What description? What do you mean?

Reply:
Is to combine all the descriptions.  You can see the response below.